R/custom-duckdb.R

```diff
@@ -0,0 +1,130 @@
+
+ensure_custom_duckdb <- function(lib = custom_duckdb_lib_dir(), install = TRUE,
+                                 quiet = FALSE) {
+  result <- tryCatch({
+      query_custom_duckdb(
+        "select scale_factor, query_nr from tpch_answers() LIMIT 1;",
+        lib = lib
+      )
+    },
+    error = function(e) {
+      error_is_from_us <- grepl(
+        "(name tpch_answers does not exist)|(there is no package called 'duckdb')",
+        conditionMessage(e)
+      )
+
+      if (error_is_from_us) {
+        NULL
+      } else {
+        rlang::abort(
+          "An unexpected error occured whilst querying TPC-H enabled duckdb",
+          parent = e
+        )
+      }
+    }
+  )
+
+  if (identical(result$query_nr, 1L)) {
```

Comment:
```suggestion
  # Check that the result has a query in it
  if (identical(result$query_nr, 1L)) {
```

Or some other kind of classificatory statement?